rllib/models/specs/specs_dict.py

```diff
@@ -107,16 +110,24 @@ def validate(
         data_keys_set = set(data.keys())
         missing_keys = self._keys_set.difference(data_keys_set)
         if missing_keys:
-            raise ValueError(_MISSING_KEYS_FROM_DATA.format(missing_keys))
+            raise ValueError(
+                _MISSING_KEYS_FROM_DATA.format(missing_keys, data_keys_set)
+            )
         if exact_match:
             data_spec_missing_keys = data_keys_set.difference(self._keys_set)
             if data_spec_missing_keys:
                 raise ValueError(_MISSING_KEYS_FROM_SPEC.format(data_spec_missing_keys))
 
         for spec_name, spec in self.items():
             data_to_validate = data[spec_name]
-            if isinstance(spec, TensorSpecs):
-                spec.validate(data_to_validate)
+            if isinstance(spec, TensorSpec):
```

Comment:
for safety, this needs an "else" clause that just raises error if spec is some random data?

Reply:
fixed.